test/ng/compileSpec.js

```diff
@@ -5629,6 +5629,34 @@ describe('$compile', function() {
       });
 
 
+      //see issue https://github.com/angular/angular.js/issues/12936
+      it('should use the proper scope when being the root element of a replaced directive', function() {
```

Comment:
I don't think it's necessary, this bug didn't appear for `templateUrl`, since it loaded and compiled the template separately.